src/resources/ECForms.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ *  @module Resources/ECForms
+ */
+import { AuthParams, ResponseCallback, SendData } from "../api/RestAPI";
+
+import { AmountWithCurrency } from "./common/types";
+import { CRUDResource } from "./CRUDResource";
+import { InstallmentPlan, SubscriptionPeriod } from "./Subscriptions";
+import { TransactionTokenType, UsageLimit } from "./TransactionTokens";
+
+type BaseMetadata = Record<string, string>;
+
+export enum CheckoutType {
+    PAYMENT = "payment",
+    TOKEN = "token",
+}
+
+export enum ECFormCustomFieldType {
+    STRING = "string",
+    SELECT = "select",
+}
+
+export type ECFormCustomField = {
+    key: string;
+    label: string;
+    type: ECFormCustomFieldType;
+    required?: boolean;
+    options?: string[];
+};
+
+export type ECFormItem<Metadata = BaseMetadata> = {
+    /* EC Form data */
+    id: string;
+    storeId: string;
+    name: string;
+    appTokenId: string;
+    createdOn: string;
+
+    /* Redirects */
+    successRedirectUrl?: string;
+    failureRedirectUrl?: string;
+
+    /* Charge and token data */
+    appId: string;
+    checkout: CheckoutType;
+    paymentType: PaymentShippingType;
+    tokenType: TransactionTokenType;
+    univapayCustomerId: string | null;
+    capture: boolean;
+    captureAt?: string | null;
+    onlyDirectCurrency: boolean;
+
+    /* Display */
+    title: string | null;
+    description: string | null;
+    confirmationRequired: boolean;
+    locale: string | null;
+    header: string | null;
+    dark: boolean;
+    submitButtonText?: string | null;
+    showCvv: boolean;
+    displayStoreName: boolean;
+    displayStoreLogo: boolean;
+    headerColor?: string;
+    buttonColor?: string;
+    backgroundColor?: string;
+
+    /* Subscription */
+    subscriptionId?: string | null;
+    subscriptionPeriod?: SubscriptionPeriod | null;
+    subscriptionInitialAmount?: AmountWithCurrency | null;
+    subscriptionStart?: string | null;
+    installmentPlan?: InstallmentPlan | null;
+    installmentQty?: number | null;
+    installmentAmount?: AmountWithCurrency | null;
+    subscriptionTimezone?: string | null;
+    subscriptionPreserveEndOfMonth?: boolean | null;
+
+    /* Recurring token */
+    usageLimit?: UsageLimit | null;
+    cvvAuthorize: boolean;
+
+    /* Address */
+    address?: boolean | null;
+    requireName?: boolean | null;
+    requireEmail?: boolean | null;
+    requireBillingAddress?: boolean | null;
+    email?: string | null;
+    shippingAddressLine1: string | null;
+    shippingAddressLine2: string | null;
+    shippingAddressCity: string | null;
+    shippingAddressState: string | null;
+    shippingAddressZip: string | null;
+    shippingAddressCountryCode: string | null;
+    buyerName: string | null;
+    buyerNameTransliteration: string | null;
+    buyerDateOfBirth: string | null;
```

Comment:
what's the difference between the usage of `xxx?: type` and `xxx: type | null`?

Reply:
Here is a list of the different combination 🙂 

```tsx
required: Type; // Type
optional?: Type; // Type | undefined
nullable: Type | null // Type | null
optionalNullable?: Type | null // Type | null | undefind
```

`null` means that the property can be `null` and `undefined` means that the property may be `undefined` or not in the object.